### PR TITLE
Add note to docs on how to override privileges

### PR DESCRIPTION
**Pull request by LoneWolfHT**
## Goal of the PR
Add some info on how to override the values of already registered privileges
## Does it resolve any reported issue?
Not reported but a few modders have had questions about this on the unofficial Discord server (Some may have asked on other platforms but I'm not as active on those)

## To do

Ready for Review.

## How to test

* [optional] Get your glasses (They're on your head before you ask)
* Look at the doc changes made by this PR

### CODE REVIEW

**Comment by sfan5** [comment]
This applies to many of the `minetest.registered_*` so adding it to just one is kinda ... incomplete.
I consider this to be obvious, but if it has to be explained it should rather be a general explanation of which / how you can modify and or override existing nodes/entities/privileges/...

**Comment by appgurueu** [comment]
> I consider this to be obvious

It is not obvious. One expects that setting `{priv = false}` - rather than `{priv = nil}` revokes a priv.

**Comment by LoneWolfHT** [comment]
> This applies to many of the `minetest.registered_*` so adding it to just one is kinda ... incomplete.
> I consider this to be obvious, but if it has to be explained it should rather be a general explanation of which / how you can modify and or override existing nodes/entities/privileges/...

Which ones does it apply to? I'm willing to add those

**Comment by rubenwardy** [comment]
>  a few modders have had questions about this on the unofficial Discord server

A few? One

**Comment by SmallJoker** [comment]
In-game editing:

 * entities -> defaults for newly initialized ones
 * privs -> handled mostly in Lua, but adding/removing requires C++ function calls
 * chatcommands -> handled entirely in Lua

Load-time only editing:

 * items / tools / nodes -> also refer to override_item
 * alias -> use functions from C++ only
 * recipes -> use functions from C++ only
 * abm / lbm -> modify whatever you want, however you want

**Comment by LoneWolfHT** [comment]
> > a few modders have had questions about this on the unofficial Discord server
> 
> A few? One

There have been some people in the past who've had this question too IIRC

Adding those others now...

File: doc/lua_api.txt [hunk 1]
```diff
@@ -4327,6 +4328,7 @@ Call these functions only at load time!
       by default.
     * To allow players with `basic_privs` to grant, see the `basic_privs`
       minetest.conf setting.
+    * To modify an already registered privilege just edit the value found in minetest.registered_privileges
```
Comment (SmallJoker) [inline doc/lua_api.txt:4331]:
```suggestion
    * Registered privileges can be modified directly in
      `minetest.registered_privileges`.
```

File: doc/lua_api.txt [hunk 2]
```diff
@@ -5554,6 +5556,7 @@ Global tables
     * Map of registered tool definitions, indexed by name
 * `minetest.registered_entities`
     * Map of registered entity prototypes, indexed by name
+    * You can use this to change the values of already registered entities
```
Comment (SmallJoker) [inline doc/lua_api.txt:5559]:
```suggestion
    * Values in this table may be modified directly.
      Note: the new values will only affect newly spawned entities.
```

File: doc/lua_api.txt [hunk 3]
```diff
@@ -5584,6 +5587,7 @@ Global tables
     * Map of registered chat command definitions, indexed by name
 * `minetest.registered_privileges`
     * Map of registered privilege definitions, indexed by name
+    * You can use this to change the values of already registered privileges
```
Comment (SmallJoker) [inline doc/lua_api.txt:5590]: How to use this? Superfluous `already`.

```suggestion
    * Registered privileges can be modified directly in this table.
```

File: doc/lua_api.txt (hunk 1 above)
Comment (sfan5) [reply doc/lua_api.txt:4331]: Please also apply this wording in other places, it's better.

**Comment by LoneWolfHT** [comment]
How does `Note: the new values will only affect entities spawned afterwards` sound?

**Comment by rubenwardy** [comment]
`Changing initial properties will only`...

File: doc/lua_api.txt [hunk 4]
```diff
@@ -5616,6 +5621,8 @@ Global tables
     * Map of registered tool definitions, indexed by name
 * `minetest.registered_entities`
     * Map of registered entity prototypes, indexed by name
+    * Values in this table may be modified directly.
+      Note: the new values will only affect entities spawned afterwards
```
Comment (rubenwardy) [inline doc/lua_api.txt:5625]: This is not true - it uses metatables, so the entity table will immediately change for all entities. However, initial properties are only read when entities are spawned

```suggestion
      Note: changes to initial properties will only affect entities spawned afterwards,
      as they are only read when spawning.
```

**Comment by paramat** [comment]
I think this is excessively over-explaining and has unnecessary duplication.
It seems wrong to add documentation about modifying a registered thing in the 'register_thing' section, the 'Global Tables' section seems the right place for this. This would also mean no duplication.

What is being added here seems rather obvious: "registered things can be modified in this table of the registered things".
So this duplication is even more unjustified.
:-1: unless de-duplicated, please only add to the 'Global Tables' section. Then i might +1 and merge this.

**Comment by LoneWolfHT** [comment]
How's that?

**Comment by paramat** [comment]
:+1: thanks.

**Pull request merged**